lib/web_ui/test/engine/pointer_binding_test.dart

```diff
@@ -1080,6 +1089,365 @@ void main() {
+      expect(packets[0].data[0].change, equals(ui.PointerChange.up));
+      expect(packets[0].data[0].synthesized, equals(false));
+      expect(packets[0].data[0].buttons, equals(0));
+      packets.clear();
+    },
+  );
+
+  _testEach<_ButtonedEventMixin>(
+    [_PointerEventContext(), _MouseEventContext()],
+    'correctly handles two consecutive RMB clicks with up in between',
+    (_ButtonedEventMixin context) {
+      PointerBinding.instance.debugOverrideDetector(context);
+      // This can happen with the following gesture sequence:
+      //
+      //  - Pops up the context menu by right clicking, but doesn't hold RMB;
+      //  - Clicks RMB again in a different location;
+      //
+      // This seems to be happening sometimes when using RMB on the Mac trackpad.
+
+      List<ui.PointerDataPacket> packets = <ui.PointerDataPacket>[];
+      ui.window.onPointerDataPacket = (ui.PointerDataPacket packet) {
+        packets.add(packet);
+      };
+
+      // Press RMB, popping up the context menu.
+      glassPane.dispatchEvent(context.mouseDown(
+        button: 2,
+        buttons: 2,
+        clientX: 10.0,
+        clientY: 10.0,
+      ));
+      expect(packets, hasLength(1));
+      expect(packets[0].data, hasLength(2));
+      expect(packets[0].data[0].change, equals(ui.PointerChange.add));
+      expect(packets[0].data[0].synthesized, equals(true));
+      expect(packets[0].data[1].change, equals(ui.PointerChange.down));
+      expect(packets[0].data[1].synthesized, equals(false));
+      expect(packets[0].data[1].buttons, equals(2));
+      packets.clear();
+
+      // RMB up.
+      glassPane.dispatchEvent(context.mouseUp(
+        button: 2,
+        clientX: 10.0,
+        clientY: 10.0,
+      ));
+      expect(packets, hasLength(1));
+      expect(packets[0].data, hasLength(1));
+      expect(packets[0].data[0].change, equals(ui.PointerChange.up));
+      expect(packets[0].data[0].synthesized, equals(false));
+      expect(packets[0].data[0].buttons, equals(0));
+      packets.clear();
+
+      // Press RMB again. In Chrome, when RMB is clicked again while the
+      // contextmenu is still active, it sends a pointerdown/mousedown event
+      // with "buttons:0".
+      glassPane.dispatchEvent(context.mouseDown(
+        button: 2,
+        buttons: 0,
+        clientX: 20.0,
+        clientY: 20.0,
+      ));
+      expect(packets, hasLength(1));
+      expect(packets[0].data, hasLength(2));
+      expect(packets[0].data[0].change, equals(ui.PointerChange.hover));
+      expect(packets[0].data[0].synthesized, equals(true));
+      expect(packets[0].data[0].buttons, equals(2));
+      expect(packets[0].data[1].change, equals(ui.PointerChange.down));
+      expect(packets[0].data[1].synthesized, equals(false));
+      expect(packets[0].data[1].buttons, equals(2));
+      packets.clear();
+
+      // Release RMB.
+      // This won't send the "up" event to the framework because the previous
```

Comment:
This is an outdated comment (good catch!). I'll remove it.

Explanation: the old implementation caused `_ButtonSanitizer.sanitizeUpEvent()` to skip this "up" event because the previous "down" had `buttons:0` which sets `_pressedButtons` to 0.

With the new implementation, that's not the case anymore. `_ButtonSanitizer.sanitizeDownEvent()` now tries to infer the value of `buttons` from `event.button`. So `_pressedButtons` is not set to zero anymore.